api/config/v1alpha1/envoygateway_types.go

```diff
@@ -68,9 +68,9 @@ type Gateway struct {
 	ControllerName string `json:"controllerName,omitempty"`
 }
 
-// Provider defines the desired configuration of a provider.
+// GatewayProvider defines the desired configuration of a provider.
 // +union
-type Provider struct {
+type GatewayProvider struct {
```

Comment:
should this be `EnvoyGatewayProvider` ? since the data plane provider is called `EnvoyProxy`

Reply:
done @arkodg